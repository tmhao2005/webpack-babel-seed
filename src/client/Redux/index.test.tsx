import React from "react";
import { shallow } from "enzyme";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import Sidebar from ".";

describe("Testing Sidebar component", () => {
  let wrapper, store;
  const mockStore = configureStore();
  const initialProp = {
    auth: {
      userAuth: {
        firstName: "savitha",
        lastName: "goel",
      },
    },
    meta: {
      productsMeta: [
        { label: "CX Admin Application" },
        { label: "Application Configuration Management" },
      ],
    },
  };

  beforeEach(() => {
    store = mockStore(initialProp);
    wrapper = shallow(<Sidebar store={store} />);
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("renders correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("render sidebar options as a text", () => {
    // This test case gets failed every time
    render(<Sidebar store={store} />);

    const sidebarElement = screen.getByText("Application Manager");
    expect(sidebarElement).toBeInTheDocument();
  });
});
