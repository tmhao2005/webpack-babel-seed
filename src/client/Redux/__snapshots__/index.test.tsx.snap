// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Sidebar component renders correctly 1`] = `
<ContextProvider
  value={null}
>
  <Sidebar
    dispatch={[Function]}
    firstName="savitha"
    lastName="goel"
    products={
      Array [
        Object {
          "label": "CX Admin Application",
        },
        Object {
          "label": "Application Configuration Management",
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
